grpcbp/prometheus.go

```diff
@@ -100,3 +102,16 @@ var (
 		Help: "The number of in-flight requests",
 	}, clientActiveRequestsLabels)
 )
+
+// serviceAndMethodSlug splits the UnaryServerInfo.FullMethod and returns
+// the package.service part separate from the method part.
+// ref: https://pkg.go.dev/google.golang.org/grpc#UnaryServerInfo
+func serviceAndMethodSlug(fullMethod string) (service string, method string) {
+	split := strings.SplitN(fullMethod, "/", 3)
+	if len(split) < 2 {
+		return "", ""
+	}
+	method = split[len(split)-1]
```

Comment:
since you already checked length, at this point the only possible `len(split)` is 3, so you can just use `method = split[2]` here.

Reply:
😅 actually len can be 2 or 3

Reply:
you are right